driver-core/src/test/functional/com/mongodb/internal/operation/AsyncQueryBatchCursorFunctionalSpecification.groovy

```diff
@@ -306,6 +307,29 @@ class AsyncQueryBatchCursorFunctionalSpecification extends OperationFunctionalSp
         false     | 0
     }
 
+    @Slow
+    def 'should unblock if closed while waiting for more data from tailable cursor'() {
+        given:
+        collectionHelper.create(collectionName, new CreateCollectionOptions().capped(true).sizeInBytes(1000))
+        collectionHelper.insertDocuments(new DocumentCodec(), Document.parse('{}'))
+        def firstBatch = executeQuery(new BsonDocument('_id', BsonNull.VALUE), 0, 1, true, true);
```

Comment:
`executeQuery` issues the following command:
```
{"find": "cappedCollectionName", "filter": {"_id": null}, "tailable": true, "awaitData": true, "limit": 0, "batchSize": 1, "$db": "JavaDriverTest", "lsid": {"id": {"$binary": {"base64": "Bfg1KfxTRE6SZBNt0yKANQ==", "subType": "04"}}}}
```
Surprisingly, if before it is issued, the collection contains 0 documents (the line 314 is commented out), then the server does not create a cursor as a result of executing the command. I looked at https://docs.mongodb.com/manual/reference/command/find/, https://docs.mongodb.com/manual/core/capped-collections/, https://docs.mongodb.com/manual/core/tailable-cursors/, and the only possibly related piece of documentation is from the last link: _"Tailable cursors may become dead, or invalid, if either: the query returns no match."_ However, the filter `{"_id": null}` does not match any documents regardless of whether the collection contains any or not.

@jyemin, @rozza do you happen to know the explanation for this behavior?

Reply:
I think it will be covered by the documented caveat. There looks to be an old server ticket: [SERVER-13955](https://jira.mongodb.org/browse/SERVER-13955)

Reply:
Ah, so it's a known bug. Thank you!